rust/noosphere-storage/src/implementation/rocks_db.rs

```diff
@@ -0,0 +1,158 @@
+use crate::{storage::Storage, store::Store, SPHERE_DB_STORE_NAMES};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use rocksdb::{ColumnFamilyDescriptor, DBWithThreadMode, Options};
+use std::{path::PathBuf, sync::Arc};
+
+#[cfg(not(feature = "rocksdb-mt"))]
+type DbInner = DBWithThreadMode<rocksdb::SingleThreaded>;
+#[cfg(not(feature = "rocksdb-mt"))]
+type ColumnType<'a> = &'a rocksdb::ColumnFamily;
+#[cfg(feature = "rocksdb-mt")]
+type DbInner = DBWithThreadMode<rocksdb::MultiThreaded>;
+#[cfg(feature = "rocksdb-mt")]
+type ColumnType<'a> = Arc<rocksdb::BoundColumnFamily<'a>>;
+
+/// A RocksDB implementation of [Storage].
+///
+/// Caveats:
+/// * Values are limited to 4GB(?) [https://github.com/facebook/rocksdb/wiki/Basic-Operations#reads]
+/// * Potential workarounds of cf_handles not being [Sync] [https://github.com/rust-rocksdb/rust-rocksdb/issues/407]
```

Comment:
`Unshared` in #624

Reply:
This has some challenges migrating to `Unshared<ColumnType<'a>>`, tracking a lifetime through a `Store` implementor (or to its relative `DbInner` instance), updating docs to add a note of Unshared